great_expectations/rule_based_profiler/attributed_resolved_metrics.py

```diff
@@ -56,49 +52,35 @@ def _detect_illegal_array_type_or_shape(values: MetricValues) -> bool:
         ):
             return True
 
+        value: MetricValue
         if isinstance(values, (list, tuple)):
             if len(values) > 0:
-                value: MetricValue = values[0]
-                # Python "None" is illegal as candidate for conversion into "numpy.ndarray" type.
-                if value is None:
-                    return True
-
-                # Pandas "DataFrame" and "Series" are illegal as candidates for conversion into "numpy.ndarray" type.
-                if isinstance(
-                    value,
-                    deep_filter_properties_iterable(
-                        properties=(
-                            pd.DataFrame,
-                            pd.Series,
-                            sqlalchemy_engine_Row,
-                            pyspark_sql_Row,
-                            set,
-                        )
-                    ),
-                ):
-                    return True
-
+                values_iterator: Iterator
                 # Components of different lengths cannot be packaged into "numpy.ndarray" type (due to undefined shape).
                 if all(isinstance(value, (list, tuple)) for value in values):
-                    values_iterator: Iterator = iter(values)
-                    first_value_length: int = len(next(values_iterator))
-                    current_value: Sized[Any]
+                    values = deep_filter_properties_iterable(properties=values)
```

Comment:
Can you annotate `values`? I think it can be `None` based on the annotated return type of `deep_filter_properties_iterable` in which case `len` will die.

Reply:
@billdirks `values` is already attributed (method argument).  Thanks.

Reply:
@billdirks I think we can leave metric_value as same type, because the MetricValues in the argument covers the return type of the filter applied.  Thanks.

Reply:
```MetricValues = Union[
    MetricValue, List[MetricValue], Set[MetricValue], Tuple[MetricValue, ...]
]```
but
```def deep_filter_properties_iterable(...) -> Union[dict, list, set, tuple, None]```
So it’s changing from something more specific a List[MetricValue] to something less specific.

Reply:
Also notice the return value can be `None`. This means the next line can crash.